### DISCUSSION TITLE
Looking for a stable release

### DISCUSSION THREAD

**Discussion by wimgaethofs** [PowerShell/Win32-OpenSSH#2136] [Q&A]
Is there no longer a stable release available? I can only find BETA releases. 
What should I run on my Windows production environment?

**Comment by maertendMSFT** [comment]
There is no "non-beta" release that is available on GitHub.  This is only due to the level of support we are able to offer on the GitHub releases.  All "GA" releases are made through Windows and Windows Update, these versions are officially supported via Windows Support.

There are no differences in quality for a Windows vs GitHub release. All GitHub releases should be stable.

**Comment by wimgaethofs** [comment]
Thanks for clarifying this. Since Windows is far behind with it's updates , is it correct to assume that it is more secure to run the latest GitHub release instead of the Windows version?

**Comment by maertendMSFT** [comment]
The version in Windows is updated during each Windows release, if you update your Windows version you will have a more recent OpenSSH version.
If there are any high priority security fixes needed for Windows, then those are released via servicing.  Though the "most secure" of any product is generally the latest release.

**Reply by raggi** [reply]
@maertendMSFT there are multiple security fixes in both the version range of the currently shipping openssh, and in each of the beta releases since then bar one. The fix for the IO race has been in for nearly year now. I have to be frank, and would ask that you also ask this up the chain: what's the holdup?

**Reply by krisavi** [reply]
@maertendMSFT 
> if you update your Windows version you will have a more recent OpenSSH version. 

That is so false... Windows 11 ships with 8.6p1 which is years old. Even the updates have not brought any newer versions.

**Reply by raggi** [reply]
https://github.com/orgs/community/discussions/48568#discussioncomment-9116448

**Reply by ranaaqil338** [reply]
> @maertendMSFT there are multiple security fixes in both the version range of the currently shipping openssh, and in each of the beta releases since then bar one. The fix for the IO race has been in for nearly year now. I have to be frank, and would ask that you also ask this up the chain: what's the holdup?

**Reply by tgauth** [reply]
> The fix for the IO race has been in for nearly year now.

Are you referring to upstream's CVE-2024-6387? If so, please see https://github.com/PowerShell/Announcements/issues/63

**Comment by ranaaqil338** [comment]
> Is there no longer a stable release available? I can only find BETA releases. 
> What should I run on my Windows production environment?

**Comment by ranaaqil338** [comment]
> Is there no longer a stable release available? I can only find BETA releases. 
> What should I run on my Windows production environment?

**Reply by tgauth** [reply]
Windows has both the OpenSSH client and OpenSSH server as features-on-demand that can be installed via optional features

**Reply by GopiMtp** [reply]
Yes, it is available for the Windows 10 and later, what if I need to install this on older operating systems? Can I download and install the latest one from this repo ? What's your recommendation on the older version of windows, if I need to add ssh server software support.

**Comment by Tty725** [comment]
10/10/2024 | 9.8.1.0 | https://github.com/PowerShell/openssh-portable/releases/tag/v9.8.1.0
-- | -- | --
04/08/2025 | 9.8.2.0 | https://github.com/PowerShell/openssh-portable/releases/tag/v9.8.2.0
04/18/2025 | 9.8.3.0 | https://github.com/PowerShell/openssh-portable/releases/tag/v9.8.3.0



Are there binary packages available for download for these release versions? Windows 2012 does not come with built-in components.

**Reply by tgauth** [reply]
Yes, please see https://github.com/PowerShell/Win32-OpenSSH/releases

**Reply by DDrackonis** [reply]
Hey there.

First of all, I want to thank you all for the amazing work you're doing keeping this repo alive.

Second:
Is there any chance there will be an update on a stable release. I'm getting several CVEs reported on 10.0.0 2026-35414 and 2026-35385 through 2026-35388 reported and being asked for a new version.

I just want to know since I will be soon be pushed to build it myself if not

**Comment by DDrackonis** [comment]
Hey there.

First of all, I want to thank you all for the amazing work you're doing keeping this repo alive.

Second:
Is there any chance there will be an update on a stable release. I'm getting several CVEs reported on 10.0.0 2026-35414 and 2026-35385 through 2026-35388 reported and being asked for a new version.

I just want to know since I will be soon be pushed to build it myself if not
